bridge/opentracing/bridge.go

```diff
@@ -497,8 +498,9 @@ func otTagsToOTelAttributesKindAndError(tags map[string]interface{}) ([]attribut
 	for k, v := range tags {
 		switch k {
 		case string(otext.SpanKind):
-			if s, ok := v.(string); ok {
-				switch strings.ToLower(s) {
+			refValue := reflect.ValueOf(v)
```

Comment:
In order to avoid using the reflect package, how about taking the opposite approach, of turning strings into `SpanKindEnum`?

```golang
case string(otext.SpanKind):
	if s, ok := v.(string); ok {
		v = otext.SpanKindEnum(strings.ToLower(s))
	}
	if s, ok := v.(otext.SpanKindEnum); ok {
		switch s {
		case otext.SpanKindRPCClientEnum:
			kind = trace.SpanKindClient
		case otext.SpanKindRPCServerEnum:
			kind = trace.SpanKindServer
		case otext.SpanKindProducerEnum:
			kind = trace.SpanKindProducer
		case otext.SpanKindConsumerEnum:
			kind = trace.SpanKindConsumer
		}
	}
```

Reply:
Yes, this looks good to me if reflect package is not the ideal solution for this. Should I update the commit?

Reply:
Let's wait for another opinion.

Reply:
@hanyuancheung @MrAlias

Reply:
I agree that `reflect` seems out of place.  I would maybe suggest a `map[otext.SpanKindEnum]trace.SpanKind` with a default so we don't leave a kind = "".